torch/optim/lr_scheduler.py

```diff
@@ -667,18 +669,21 @@ class SequentialLR(LRScheduler):
     """
 
     def __init__(self, optimizer, schedulers, milestones, last_epoch=-1, verbose="deprecated"):
-        for scheduler_idx in range(len(schedulers)):
-            if schedulers[scheduler_idx].optimizer != optimizer:
+        if len(schedulers) < 1:
+            raise ValueError("Sequential Schedulers expects at least one scheduler, but got no scheduler.")
+
+        for scheduler_idx, scheduler in enumerate(schedulers):
+            if not isinstance(scheduler, LRScheduler):
```

Comment:
Hi @janeyx99 , thanks for pointing this out. I do agree that accepting `ReduceLROnPlateau` in `SequentialLR` or `ChainedScheduler` would be weird. However, before and after this change actually nither prevent users from passing ReduceLR in. Do you think if we explicitly raise ValueError if get `ReduceLROnPlateau` in both of the two classes be better?

Thanks~

Reply:
Ah, my point is moreso that since there was precedence where ReduceLROnPlateau was not a strict subclass, others may write their own LRSchedulers without it being a subclass, and if they're trying to plug those LRSchedulers into SequentialLR, this change would be BC breaking for that use case.

If they have this change, the concern about ReduceLROnPlateau changes since that is now a subclass and won't trigger this error.